### DISCUSSION TITLE
Oqtane 2.2 external module containing multiple modules example needed

### DISCUSSION THREAD

**Discussion by Velobook** [oqtane/oqtane.framework#1729] [Q&A] [answered]
I'm searching a solution to build a module able to call different modules from a dashboard page like the dashboard Admin Module of Oqtane.  This module will be a complete application.
I've made that a lot of with Dotnetnuke for a while. With differents controls and key controls for navigating 
How to attach different pages to the same module?  How to navigate between these pages?
I've seen discussion where you explain that we can take same architecture than the framework but not sure to understand how to do that..

**Comment by curlyfro** [comment]
i don't understand the problem.  you don't know how to put the same module on same/different pages?   sounds like you have a DNN background and are familiar with setting modules on pages.   or is your problem state management?  please try to explain what you're trying to do.

**Reply by Velobook** [reply]
Yes DNN background!
I got a nice answer from @leigh-pointer just below. 
Thanks

**Comment by leigh-pointer** [comment, marked as answer]
@Velobook simply add and new component to your module and [**NameIt**] make sure that your new component inherits from ModuleBase.
@inherits ModuleBase;

In your @code block 

@code {
    public override SecurityAccessLevel SecurityAccessLevel => SecurityAccessLevel.Edit;
    public override string Actions => "**NameIt**";
}

From your Index component add a button

@inherits ModuleBase;
@inject NavigationManager NavigationManager

<button type="button" class="btn btn-secondary" @onclick=@(e => NavigationManager.NavigateTo(EditUrl(ModuleState.ModuleId, "**NameIt**")))>

That should get you going.

**Reply by Velobook** [reply]
Thanks for your reply. It's exactly what I need.
I just  forgot the magic NavigateUrl and EditUrl  so used in DNN! 
I just replaced the button by NavLink component
Now I can go further with Oqtane. What I want to do is 
"modules acting like a single application" with their own menu. Like the admin dashboard
In my test project I want to update an application build with DNN 7.4 by a new application in Blazor. 
Oqtane is my way.
